### PR TITLE
Added a logging and monitoring architecture diagram and page

### PR DESCRIPTION
**Pull request by vikramkoka**
Added a logging and monitoring architecture diagram and page into the logging and monitoring section of the docs. 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by kaxil — APPROVED** [review]

File: docs/logging-monitoring/logging-architecture.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ .. Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+
+
+Logging and Monitoring architecture
```
Comment (mik-laj) [inline docs/logging-monitoring/logging-architecture.rst:20]: What do you think to move this section up?
https://airflow.readthedocs.io/en/latest/logging-monitoring/index.html
This page is very empty and has been waiting for someone to contribute to the introduction.

File: docs/logging-monitoring/logging-architecture.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+===================================
+
+Airflow supports a variety of logging and monitoring mechanisms as shown below.
+
+.. image:: ../img/arch-diag-logging.png
+
+By default, Airflow supports logging into the local file system. These include logs from the Web server, the Scheduler, and the Workers running tasks. This is suitable for development environments and for quick debugging.
+
+For cloud deployments, Airflow also has hooks contributed by the Community for logging to cloud storage such as AWS, GCP, and Azure.
```
Comment (mik-laj) [inline docs/logging-monitoring/logging-architecture.rst:29]: hooks => handlers or task handlers

Comment (mik-laj) [inline docs/logging-monitoring/logging-architecture.rst:29]: GCP => Google Cloud
See:  https://developers.google.com/style/word-list#gcp

File: docs/logging-monitoring/logging-architecture.rst (hunk 1 above)
Comment (vikramkoka) [reply docs/logging-monitoring/logging-architecture.rst:20]: Sure, that's a good idea. I will move it up and add to it.

File: docs/logging-monitoring/logging-architecture.rst (hunk 2 above)
Comment (vikramkoka) [reply docs/logging-monitoring/logging-architecture.rst:29]: Done, good catch

Comment (vikramkoka) [reply docs/logging-monitoring/logging-architecture.rst:29]: done

Comment (mik-laj) [reply docs/logging-monitoring/logging-architecture.rst:29]: Can you create a new PR?

Comment (vikramkoka) [reply docs/logging-monitoring/logging-architecture.rst:29]: done

**Pull request merged**